Meadowlark-js/packages/meadowlark-core/src/validation/DocumentValidator.ts

```diff
@@ -11,7 +11,7 @@ import { validateEntityBodyAgainstSchema } from '../metaed/MetaEdValidation';
  *
  * Validates the request body for the resource. If invalid, returns an error message.
  */
-export async function validateDocument(body: object, matchingMetaEdModel: TopLevelEntity): Promise<string> {
+export async function validateDocument(body: object, matchingMetaEdModel: TopLevelEntity): Promise<object | null> {
```

Comment:
Should be able to get more specific than `object`. How about defining `type ValidationFailure = { error: ValidationError[] };`